conans/test/integration/command_v2/list_recipes_test.py

```diff
@@ -0,0 +1,267 @@
+        recipe_name = "test_recipe/1.0.0@user/channel"
+        self._create_recipe(recipe_name)
+
+        self.client.run("list recipes --cache")
+        assert "{}#".format(recipe_name) in self.client.out
+
+
+class TestListRecipesFromRemotes(TestListRecipesBase):
+    def test_search_no_matching_recipes(self):
+        self._add_remote("remote1")
+        self._add_remote("remote2")
+
+        expected_output = ("Local Cache:\n"
+                           "  There are no matching recipes\n"
+                           "remote1:\n"
+                           "  There are no matching recipes\n"
+                           "remote2:\n"
+                           "  There are no matching recipes\n")
+
+        self.client.run("list recipes whatever")
+        assert expected_output == self.client.out
+
+    def test_fail_if_no_configured_remotes(self):
+        self.servers = OrderedDict()
+        self.client = TestClient(servers=self.servers)
+
+        self.client.run("list recipes whatever", assert_error=True)
+        assert "ERROR: The remotes registry is empty" in self.client.out
+
+    def test_search_disabled_remote(self):
+        self._add_remote("remote1")
+        self.client.run("remote disable remote1")
+        self.client.run("list recipes whatever -r remote1", assert_error=True)
+        assert "ERROR: Remote 'remote1' is disabled" in self.client.out
+
+
+class TestRemotes(TestListRecipesBase):
+    def test_no_remotes(self):
+        self.client.run("list recipes something", assert_error=True)
+        expected_output = "The remotes registry is empty. Please add at least one valid remote"
+        assert expected_output in self.client.out
+
+    def test_search_by_name(self):
+        remote_name = "remote1"
+        recipe_name = "test_recipe/1.0.0@user/channel"
+        self._add_remote(remote_name)
+        self._upload_recipe(remote_name, recipe_name)
+
+        self.client.run("remote list")
+
+        assert "remote1: http://fake" in self.client.out
+
+        self.client.run("list recipes -r {} {}".format(remote_name, "test_recipe"))
+
+        expected_output = (
+            "remote1:\n"
+            "  test_recipe\n"
+            "    {}\n".format(recipe_name)
+        )
+
+        assert expected_output == self.client.out
+
+    def test_search_in_all_remotes(self):
+        expected_output = (
+            r"Local Cache:\n"
+            r"  test_recipe\n"
+            r"    test_recipe/1.0.0@user/channel#.*\n"
+            r"    test_recipe/1.1.0@user/channel#.*\n"
+            r"    test_recipe/2.0.0@user/channel#.*\n"
+            r"    test_recipe/2.1.0@user/channel#.*\n"
+            r"remote1:\n"
+            r"  test_recipe\n"
+            r"    test_recipe/1.0.0@user/channel\n"
+            r"    test_recipe/1.1.0@user/channel\n"
+            r"remote2:\n"
+            r"  test_recipe\n"
+            r"    test_recipe/2.0.0@user/channel\n"
+            r"    test_recipe/2.1.0@user/channel\n"
+        )
```

Comment:
Shouldn't it be cleaner to use `textwrap` for all the `expected_output`s?
For instance:

```python3
expected_output = textwrap.dedent("""
    Local Cache:
      test_recipe
        test_recipe/1.0.0@user/channel#.*
        test_recipe/1.1.0@user/channel#.*
        test_recipe/2.0.0@user/channel#.*
        test_recipe/2.1.0@user/channel#.*
    remote1:
      test_recipe
        test_recipe/1.0.0@user/channel
        test_recipe/1.1.0@user/channel
    remote2:
      test_recipe
        test_recipe/2.0.0@user/channel
        test_recipe/2.1.0@user/channel
""")
```

Reply:
I tried, but the string is a regular expression (note the `.*` in the Local Cache recipes), and it doesn't work with textwrap (or at least I didn't manage to make it work)

Reply:
`` textwrap.dedent(r"""...)`` didn't work?

Reply:
Nope. My guess is that `dedent()` always returns a plain string, not a raw string.